src/common/snippets/src/lowered/pass/insert_specific_iterations.cpp

```diff
@@ -4,24 +4,78 @@
 
 #include "snippets/lowered/pass/insert_specific_iterations.hpp"
 
-#include "snippets/itt.hpp"
 #include "snippets/lowered/linear_ir.hpp"
-#include "snippets/lowered/loop_manager.hpp"
-#include "snippets/snippets_isa.hpp"
+#include "snippets/lowered/specific_loop_iter_types.hpp"
 #include "snippets/utils.hpp"
+#include "snippets/itt.hpp"
+
+#include <array>
 
 namespace ov {
 namespace snippets {
 namespace lowered {
 namespace pass {
 
-LinearIR::constExprIt InsertSpecificIterations::insert_copy_loop(LinearIR& linear_ir, const size_t loop_id, const LinearIR::constExprIt& insert_pos) {
+bool InsertSpecificIterations::is_decomposed_loop_needed(const UnifiedLoopInfoPtr& unified_loop_info, SpecificLoopIterType type,
+                                                         size_t remaining_work_amount) {
+    OPENVINO_ASSERT(unified_loop_info, "UnifiedLoopInfo is missed!");
+    const auto increment = unified_loop_info->get_increment();
+    OPENVINO_ASSERT(!utils::is_dynamic_value(increment) && increment > 0, "Incorrect increment: ", increment);
+    const auto is_dynamic = utils::is_dynamic_value(remaining_work_amount);
+
+    switch (type) {
+        case (SpecificLoopIterType::FIRST_ITER):
+            return !unified_loop_info->get_handlers().get_first_iter_handlers().empty() && (is_dynamic || remaining_work_amount >= increment);
+        case (SpecificLoopIterType::MAIN_BODY):
+            return is_dynamic || remaining_work_amount >= increment;
+        case (SpecificLoopIterType::LAST_ITER):
+            return (is_dynamic && increment > 1) || (!is_dynamic && remaining_work_amount > 0);
+        default:
+            OPENVINO_THROW("Unknown SpecificLoopIterType!");
+    }
+}
+size_t InsertSpecificIterations::get_decomposed_loop_work_amount(const UnifiedLoopInfoPtr& unified_loop_info, SpecificLoopIterType type,
```

Comment:
Minor: If we make all this `get_*` function to lambdas, we will be able to capture `remaining_work_amount` and `unified_looop_info`  (and `is_dynamic`) by reference, so we won't have pass the same arguments over-and-over again (even if they have not changed, like `unified_looop_info`).
I think it will increase readability, but I do not insist.